### ISSUE TITLE
Bug: Admin User unable to create any realm file

### ISSUE THREAD

**Issue by ojarabo** [realm/realm-mobile-platform#205]
Bugs: 
I was trying to accomplish what is described [here](https://github.com/realm/realm-mobile-platform/issues/189),  and I have found that the problem is not that the admin user is no able to create a global Realm file, the admin is not even able to create any realm file of any kind. 

## Goals
1)Create a Global realm file (/default) logged with the admin user to later give write/read permissions to all users or 2) a Realm file (/~/default) logged with the admin user to later offer permissions to all the users. 

## Expected Results
Create the Realm files

## Actual Results
Admin was not able to create non of the file mentioned above
## Steps to Reproduce
1) Use the Admin user you create when entering first time in the ROS, or create a new user and give admin permission in the ROS. 
2) use the email/password credential to log in the app
3) try to create a realm file (global, or user based)

## Code Sample

I am using the following **URLs** to connect:
```java
public static final String AUTH_URL         = "http://" + BuildConfig.OBJECT_SERVER_IP + ":9080/auth";
public static final String USER_REALM_URL   = "realm://" + BuildConfig.OBJECT_SERVER_IP + ":9080/~/default";
 public static final String COMMON_REALM_URL = "realm://" + BuildConfig.OBJECT_SERVER_IP + ":9080/default";
```
The user is logging either by facebook, google, or email/password. 
**Email/pass**
```java
//if the user has signed up before
SyncUser.loginAsync(SyncCredentials.usernamePassword(email, password, false), AUTH_URL, this);
@Override
    public void onSuccess(SyncUser syncUser) {
UserManager.setActiveUser(syncUser);
}

//New user
SyncUser.loginAsync(SyncCredentials.usernamePassword(eMail, password, true), AUTH_URL, new SyncUser.Callback(){
 @Override
                public void onSuccess(SyncUser syncUser) {
                    UserManager.setActiveUser(syncUser);
                }
......
 public static void setActiveUser(SyncUser user) {
        Realm.removeDefaultConfiguration();
        SyncConfiguration commonRealmConfiguration = new SyncConfiguration.Builder(user, USER_REALM_URL).build();
        Realm.setDefaultConfiguration(commonRealmConfiguration);
    }

```
## Version of Realm and Tooling

- Realm Object Server Version: ?
- Flavor:
  - [x] Developer
  - [ ] Professional
  - [ ] Enterprise
- Server OS & Version: ? (e.g. CentOS 6) Ubuntu 16.04, Realm Object Server v1.7.2
- Client SDK Version: ? Android Studio 2.3.2 (compileSdkVersion 25), Realm 3.3.1
- Client OS & Version: ?

## Logs

<details>
  <summary>systemctl status realm-object-server.service</summary>
  <code>
    realm-object-server.service - Realm Object Server
   Loaded: loaded (/etc/systemd/system/realm-object-server.service; enabled; ven
   Active: active (running) since lun 2017-05-29 11:17:03 CEST; 7h ago
 Main PID: 1084 (realm-object-se)
   CGroup: /system.slice/realm-object-server.service
           ├─1084 /usr/lib/realm-object-server-developer/node/bin/node /usr/bin/
           ├─1636 statsd {"flushInterval":10000,"flush":true}                   
           └─1648 /usr/lib/realm-object-server-developer/node/bin/node /usr/lib/

may 29 11:17:03 mycomputer systemd[1]: Started Realm Object Server.
may 29 11:17:22 mycomputer realm-object-server[1084]: 2017-05-29T09:17:22.562Z
  </code>
</details>

<details>
  <summary>sudo journalctl -u realm-object-server.service</summary>
  <code>
    -- Logs begin at lun 2017-05-29 11:16:35 CEST, end at lun 2017-05-29 18:45:51 CE
may 29 11:17:03 mycomputer systemd[1]: Started Realm Object Server.
may 29 11:17:22 mycomputer realm-object-server[1084]: 2017-05-29T09:17:22.562Z
  </code>
</details>

<details>
  <summary>cat /var/log/realm-object-server.log</summary>
  <code>
    2017-05-29T16:16:21.887Z - warn: [proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0
    at Object.parse (native)
    at ProxyService.webSocketUpgradeHandler (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:104:30)
    at ProxyService.safeUpgrade (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:78:12)
    at emitThree (events.js:116:13)
    at Server.emit (events.js:194:7)
    at onParserExecuteCommon (_http_server.js:411:14)
    at HTTPParser.onParserExecute (_http_server.js:379:5)
2017-05-29T16:16:22.791Z - warn: [proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0
    at Object.parse (native)
    at ProxyService.webSocketUpgradeHandler (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:104:30)
    at ProxyService.safeUpgrade (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:78:12)
    at emitThree (events.js:116:13)
    at Server.emit (events.js:194:7)
    at onParserExecuteCommon (_http_server.js:411:14)
    at HTTPParser.onParserExecute (_http_server.js:379:5)
2017-05-29T16:44:29.842Z - warn: [proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0
    at Object.parse (native)
    at ProxyService.webSocketUpgradeHandler (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:104:30)
    at ProxyService.safeUpgrade (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:78:12)
    at emitThree (events.js:116:13)
    at Server.emit (events.js:194:7)
    at onParserExecuteCommon (_http_server.js:411:14)
    at HTTPParser.onParserExecute (_http_server.js:379:5)
2017-05-29T16:44:30.822Z - warn: [proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0
    at Object.parse (native)
    at ProxyService.webSocketUpgradeHandler (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:104:30)
    at ProxyService.safeUpgrade (/usr/lib/nodejs/realm-object-server-developer/.build/src/node/services/proxy.js:78:12)
    at emitThree (events.js:116:13)
    at Server.emit (events.js:194:7)
    at onParserExecuteCommon (_http_server.js:411:14)
    at HTTPParser.onParserExecute (_http_server.js:379:5)

  </code>
</details>

**Comment by mokkun** [comment]
This is related to https://github.com/realm/realm-mobile-platform/issues/202

**Comment by bigfish24** [comment]
Sorry about this. Can you please post the code that is creating/opening the Realm files (both the global and the `/~/userSpecfic`) and then the permission changing code as well? Thanks!

**Comment by ojarabo** [comment]
Hi @bigfish24,
Thanks for you answer. 
These are the following URLs for creating both realm files:
```java
public static final String COMMON_REALM_URL = "realm://" + BuildConfig.OBJECT_SERVER_IP + ":9080/" + REALM_NAME;
public static final String USER_REALM_URL   = "realm://" + BuildConfig.OBJECT_SERVER_IP + ":9080/~/" + REALM_NAME;
```

This is the function that is responsible for creating the file:
```java
 public static void setActiveUser(SyncUser user) {
        Realm.removeDefaultConfiguration();
        SyncConfiguration commonRealmConfiguration = new SyncConfiguration.Builder(user, COMMON_REALM_URL).build();
        Realm.setDefaultConfiguration(commonRealmConfiguration);
    }
```

**Comment by mrackwitz** [comment]
Further investigations confirm that this is a bug in the Java client SDK. We're working on it. Please refer to https://github.com/realm/realm-java/issues/4750 for more info.

**Comment by cmelchior** [comment]
Hi @ojarabo 
Can you clarify how you are logging in the admin user? The code you posted seem to just login a normal user?

**Comment by cmelchior** [comment]
This has been fixed with Realm Java 3.3.2
